src/compiler/crystal/syntax/parser.cr

```diff
@@ -5621,6 +5621,12 @@ module Crystal
               arg_type = parse_bare_proc_type
               skip_space_or_newline
 
+              args.each do |arg|
+                if arg.name == arg_name
+                  raise "duplicated fun parameter name: #{arg_name}", arg_location
+                end
+              end
```

Comment:
I don't know if using each is an optimization but isn't there a built in method for this?

```suggestion
              if args.includes?(arg_name)
                raise "duplicated fun parameter name: #{arg_name}", arg_location
              end
```

Reply:
Would need to do something like:

```cr
if args.any? &.name.==(arg_name)
  raise "duplicated fun parameter name: #{arg_name}", arg_location
end
```

since `args` is an array of objects and not just a simple array of strings or something.

Reply:
This was copied from the check for defs which also needed to look at the external parameter names (`any?` would require two iterations as the error messages are different).

I don't think `any?` here is necessarily cleaner than the current approach.